hack/deploy-sprayproxy.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# ------------------------------------------------------------------------------------
+# Spray Proxy that is used for distrubuting Github App requests to ephemeral clusters
+# ------------------------------------------------------------------------------------
+SPRAYPROXY_QE_TEM_DIR=/tmp/prayproxy-qe
+
+echo
+echo "Deploying SprayProxy on the cluster"
+rm -rf ${SPRAYPROXY_QE_TEM_DIR} 2>/dev/null || true
+git clone --depth=1 https://github.com/xinredhat/sprayproxy-qe.git ${SPRAYPROXY_QE_TEM_DIR}
```

Comment:
we should not use personal git repository. But anyway it would be better to create development overlay in https://github.com/redhat-appstudio/infra-deployments/tree/main/components/sprayproxy and deploy in common way.

Reply:
@Michkov this Sprayprox is not part of RHTAP, which is used as a proxy  between the Github App and test cluster to distribute GithuApp webhook's requests to Pac running on the test cluster. It doesn't need to deploy this Sprayproxy when launching RHTAP CI, as there is a permeant Sprayproxy server. So  it doesn't make sense to create development overlay in https://github.com/redhat-appstudio/infra-deployments/tree/main/components/sprayproxy . Can we put it to other repo under a particular Github org?

Reply:
Not seeing reason to develop something by RHTAP team which is not part of RHTAP. Global servers could be issues in general and my understading was that we are trying to avoid it. Also when the "qe" sprayproxy would be updated then it could break all the testing. I do not see reason why to handle SprayProxy as standalone server outside of RHTAP deployment.

CC @sbose78 @ralphbean @arewm @adambkaplan - WDYT?

Reply:
@xinredhat Sprayproxy is most definitely part of our current "multi-cluster" RHTAP solution. If you are deploying RHTAP with a "host cluster" for namespace/workspace management, Sprayproxy is used to forward Github webhooks to member clusters.

Reply:
@adambkaplan  This Sprayproxy deployment is for local testing purpose.   There is a Sprayproxy running on a permanent OCP cluster, which is used to register and unregister backend clusters on fly to test RHTAP on multiple ephemeral clusters in e2e CI. When debugging Pac tests locally, the preview.sh will deploy this Sprayproxy on cluster. I don't think we can deploy it via  argocd.